### PR TITLE
Added idle_timeout setting to job settings

### PR DESCRIPTION
**Pull request by x0r3ax3ax**
##### SUMMARY

#10829 
##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

 - API
 - UI

##### AWX VERSION

```
19.2.2
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Review by nixocio — APPROVED** [review]
UI changes are sane. Thanks @oweel.

File: awx/main/conf.py [hunk 1]
```diff
@@ -408,6 +408,18 @@
     unit=_('seconds'),
 )
 
+register(
+    'DEFAULT_JOB_IDLE_TIMEOUT',
+    field_class=fields.IntegerField,
+    min_value=0,
+    default=600,
```
Comment (jakemcdermott) [inline awx/main/conf.py:415]: Is this also the default value used by runner for `idle_timeout`?

If we add this setting using a `default` of `600`, will impact the existing behavior of preexisting job templates after upgrading?

Comment (jakemcdermott) [inline awx/main/conf.py]: Is this also the default value used by runner for idle_timeout?

Will the behavior of preexisting job templates be impacted after upgrading?

Comment (x0r3ax3ax) [reply awx/main/conf.py]: @jakemcdermott thank you for your feedback! 
Yes, you are right. I set default value is 0 for prevent overriding idle_timeout in preexisting job templates.

**Comment by felipe4334** [comment]
@shanemcd It would be great if you could please review this pull request.

**Comment by felipe4334** [comment]
Hello @shanemcd

**Comment by felipe4334** [comment]
@nixocio Can this please be merged?

**Comment by felipe4334** [comment]
@oweel it looks like idle_timeout might not be working too well.

Even though idle_timeout is set to 0 its still timing out in 5-6 minutes.
image

image

- pause:
      minutes: 20

If idle_timeout is set to something like 10 seconds then it works.
image
image


And if idle_timeout is set to like 9999 it will still timeout in 5-6 minutes:
image
image

**Pull request merged**